src/component.js

```diff
@@ -111,7 +111,7 @@ export function getDomSibling(vnode, childIndex) {
 			// Since updateParentDomPointers keeps _dom pointer correct,
 			// we can rely on _dom to tell us if this subtree contains a
 			// rendered DOM node, and what the first rendered DOM node is
-			return sibling._dom;
+			return sibling._component ? sibling._component.base : sibling._dom;
```

Comment:
Hi! (Popping in from vacation 🌴 for a moment 😁. I have limited internet so I'll be away, so don't wait for me)

My thinking when playing with this method last was that `_dom` was the source of truth and that `c.base` was just a reflection of that value (perhaps one day it is a getter property). Is there a situation where `v._component.base != v._dom`? Perhaps we should fix the code that updates `c.base` and not `v._dom` (assuming my understanding is anywhere near close to meaningful 😬)?

Reply:
I'll try and debug it again, last time when debugging this I put `debugger;` everywhere the _dom got reset but this resulted in no triggers. Still, the `_dom` was set correctly when SCU returned false but then when it did have to update the `_dom` magically turned into `null`. This might be something with a ref somewhere but yes, did not find anything that made sense, I'll try and take a look at it as mentioned but can't promise anything.

Reply:
I've spent the last 2 hours debugging this again and it's really odd... Some mutation is goofing up this behavior since c.base is correct and even here https://github.com/preactjs/preact/pull/1888/files#diff-44070032789f10581d1b70740a2aaacfR93

The value is correct but somewhere it gets mutated, I even think it's mutated as a part newVNode and reflected on oldVNode due to us treating `.dom` as a mutable thing...